molecule-runtime/src/javaMain/kotlin/app/cash/molecule/platform.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (C) 2025 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.molecule
+
+internal actual fun defaultSnapshotNotifier(): SnapshotNotifier {
+  return System.getProperty("app.cash.molecule.snapshotNotifier")
+    ?.let { property ->
+      runCatching { SnapshotNotifier.valueOf(property) }.getOrNull()
```

Comment:
Any thoughts about how we would safely make migrations to this enum in the future that don't lead to runtime crashes when people update? Just rely on existing binary compatibility practices/tests?

Reply:
Yeah, you really can't. They're set in stone forever.